pkg/util/ipset/ipset.go

```diff
@@ -169,6 +169,11 @@ func (e *Entry) Validate(set *IPSet) bool {
 		return false
 	}
 	switch e.SetType {
+	case HashIP:
+		//check if IP and Protocol of Entry is valid.
+		if valid := e.checkIPandProtocol(set); !valid {
```

Comment:
`hash:ip` has no protocol

Reply:
(this comment still applies... it looks like `checkIPandProtocol()` will accept an `Entry` with no `Protocol`, but it will fill in a default value of `"TCP"`, which isn't really what you want... there ought to be a `checkIP()` function that doesn't do that...)